src/rfc3986/validators.py

```diff
@@ -411,7 +438,9 @@ def subauthority_component_is_valid(uri, component):
         return True
 
     try:
-        port = int(subauthority_dict["port"])
+        # fmt: off
```

Comment:
I won't lie, this feels really ugly, but here's my justification for it over from the commit it was done in:
> - Ignore a line being too long due to a pyright: ignore + explanation.
 - Moving the explanation away from that line doesn't make the line short enough to pass. Removing the error code as well would be enough, but removing both has 2 downsides:
  a) The lack of explanation on the same line makes it harder to immediately see why the pyright: ignore was added when grepping.
  b) Having an unrestricted pyright: ignore (or type: ignore) can cover up other typechecking errors, which could cause a problem down the line.
 - Having a pyright: ignore, a noqa, and fmt: off/on for this one line isn't clean, but I'd like a second opinion on how to handle it.
Just wanted to double-check that there isn't a better way that I'm missing, or if the justification isn't enough and the reason and error code should be shunted to a different line?

Reply:
Is this because the value for port can be None? 

If so, we can say 

```py
if (port_value := sub authority["port"]) is not None:
    try:
    ... 
    except TypeError:
    ... 
```

Because that should narrow the type to str and avoid that rule for pyright right?

Reply:
Yeah, that's possible. I wanted to avoid restructuring code just to satisfy flake8 + black, but the alternative is this collection of signal comments for the linter, formatter, and type-checker. It's not like the `except TypeError` doesn't catch everything relevant here; it's just that telling pyright to ignore the line forces me to tell flake8 to ignore the line length, which forces me to tell black to not reformat the line and move the ignore, haha.

Reply:
Actually, follow-up (inspired a bit by a convo in Discord): Could the code be restructured to avoid testing via what `int()` allows, thus avoiding the try-except in the first place? e.g. maybe something like this:

```py
if port is None:
    return True
if isinstance(port, str) and not (port.isdigit() and len(port) < 6):  # Since we know it has to be 5 digits
    return False
return 0 <= int(port) <= 65535
```

Reply:
@Sachaa-Thanasius that looks good to me.

Reply:
Cheers, it's added. @sigmavirus24.